src/phase0/state_transition/mod.rs

```diff
@@ -485,3 +556,467 @@ pub fn compute_fork_data_root(
+    const ETH1_DATA_VOTES_BOUND: usize,
+    const VALIDATOR_REGISTRY_LIMIT: usize,
+    const EPOCHS_PER_HISTORICAL_VECTOR: usize,
+    const EPOCHS_PER_SLASHINGS_VECTOR: usize,
+    const MAX_VALIDATORS_PER_COMMITTEE: usize,
+    const PENDING_ATTESTATIONS_BOUND: usize,
+>(
+    state: &BeaconState<
+        SLOTS_PER_HISTORICAL_ROOT,
+        HISTORICAL_ROOTS_LIMIT,
+        ETH1_DATA_VOTES_BOUND,
+        VALIDATOR_REGISTRY_LIMIT,
+        EPOCHS_PER_HISTORICAL_VECTOR,
+        EPOCHS_PER_SLASHINGS_VECTOR,
+        MAX_VALIDATORS_PER_COMMITTEE,
+        PENDING_ATTESTATIONS_BOUND,
+    >,
+    indices: &[ValidatorIndex],
+    context: &Context,
+) -> Result<Gwei, Error> {
+    let total_balance = indices
+        .iter()
+        .try_fold(Gwei::default(), |acc, i| {
+            acc.checked_add(state.validators[*i].effective_balance)
+        })
+        .ok_or(Error::Overflow)?;
+    Ok(u64::max(total_balance, context.effective_balance_increment))
+}
+
+pub fn get_total_active_balance<
+    const SLOTS_PER_HISTORICAL_ROOT: usize,
+    const HISTORICAL_ROOTS_LIMIT: usize,
+    const ETH1_DATA_VOTES_BOUND: usize,
+    const VALIDATOR_REGISTRY_LIMIT: usize,
+    const EPOCHS_PER_HISTORICAL_VECTOR: usize,
+    const EPOCHS_PER_SLASHINGS_VECTOR: usize,
+    const MAX_VALIDATORS_PER_COMMITTEE: usize,
+    const PENDING_ATTESTATIONS_BOUND: usize,
+>(
+    state: &BeaconState<
+        SLOTS_PER_HISTORICAL_ROOT,
+        HISTORICAL_ROOTS_LIMIT,
+        ETH1_DATA_VOTES_BOUND,
+        VALIDATOR_REGISTRY_LIMIT,
+        EPOCHS_PER_HISTORICAL_VECTOR,
+        EPOCHS_PER_SLASHINGS_VECTOR,
+        MAX_VALIDATORS_PER_COMMITTEE,
+        PENDING_ATTESTATIONS_BOUND,
+    >,
+    context: &Context,
+) -> Result<Gwei, Error> {
+    get_total_balance(
+        state,
+        &get_active_validator_indices(state, get_current_epoch(state, context)),
+        context,
+    )
+}
+
+pub fn get_indexed_attestation<
+    const SLOTS_PER_HISTORICAL_ROOT: usize,
+    const HISTORICAL_ROOTS_LIMIT: usize,
+    const ETH1_DATA_VOTES_BOUND: usize,
+    const VALIDATOR_REGISTRY_LIMIT: usize,
+    const EPOCHS_PER_HISTORICAL_VECTOR: usize,
+    const EPOCHS_PER_SLASHINGS_VECTOR: usize,
+    const MAX_VALIDATORS_PER_COMMITTEE: usize,
+    const PENDING_ATTESTATIONS_BOUND: usize,
+>(
+    state: &BeaconState<
+        SLOTS_PER_HISTORICAL_ROOT,
+        HISTORICAL_ROOTS_LIMIT,
+        ETH1_DATA_VOTES_BOUND,
+        VALIDATOR_REGISTRY_LIMIT,
+        EPOCHS_PER_HISTORICAL_VECTOR,
+        EPOCHS_PER_SLASHINGS_VECTOR,
+        MAX_VALIDATORS_PER_COMMITTEE,
+        PENDING_ATTESTATIONS_BOUND,
+    >,
+    attestation: &Attestation<MAX_VALIDATORS_PER_COMMITTEE>,
+    context: &Context,
+) -> Result<IndexedAttestation<MAX_VALIDATORS_PER_COMMITTEE>, Error> {
+    let bits = &attestation.aggregation_bits;
+    let mut attesting_indices = get_attesting_indices(state, &attestation.data, bits, context)?;
+    attesting_indices.sort_unstable();
```

Comment:
this is a detail but the spec used python's `sorted` which is a *stable* sort:

> The built-in sorted() function is guaranteed to be stable. 

https://docs.python.org/3/library/functions.html#sorted

Reply:
so to handle this i would:

1) update `get_attesting_indices` to return `HashSet`
2) still make `attesting_indices` as `List::from_iter` and then call `attesting_indices.sort()`
3) supply this value to the `IndexedAttestation`

this API is changing a bit in a PR i'm working on for `ssz_rs` but i will come back and fix that when it lands

Reply:
ok